File: 010_divisors/divisor.rb
#!/usr/bin/env ruby

total_num = cur_num = 0
ARGF.each_with_index do |line, idx|
    num = line.to_i

    if idx == 0
        total_num = num
        next
    else
        cur_num += 1
    end

    sum = 0
    for i in 1..num-1
        if num % i == 0
            sum += i
        end
    end
    puts sum 
end

File: 010_divisors/divisor.pl
#!/usr/bin/env perl

use common::sense;

my ($total, $current);
while (my $num = <>) {

    chomp $num;

    if ($. == 1) {

        $total = $num;
        next;
    }

    # process only total_num records
    last if ++$current > $total;

    my $sum;
    for (1 .. $num - 1) {

       $sum += $_ if $num % $_ == 0; 
    }
    say $sum;
}
